tests/src/unit_v2_tests.cpp

```diff
@@ -267,4 +272,84 @@ TEST(stronk_units_v2, can_override_units)
     EXPECT_EQ(back_to_b, b);
 }
 
+TEST(stronk_units_v2, scales_creates_the_right_types)
+{
+    auto m = make<meters, double>(1'000.0);
+    auto km = make<std::kilo, meters, double>(4.0);
+    auto um = make<std::micro, meters, double>(1'000.0);
+
+    static_assert(std::same_as<decltype(m), meters::value<std::ratio<1>, double>>);
+    static_assert(std::same_as<decltype(km), meters::value<std::kilo, double>>);
+    static_assert(std::same_as<decltype(um), meters::value<std::micro, double>>);
+
+    static_assert(std::same_as<decltype(m * m), meters_squared::value<std::ratio<1>, double>>);
+    static_assert(std::same_as<decltype(m * km), meters_squared::value<std::kilo, double>>);
+    static_assert(std::same_as<decltype(m * um), meters_squared::value<std::micro, double>>);
+
+    static_assert(std::same_as<decltype(km * km), meters_squared::value<std::mega, double>>);
+    static_assert(std::same_as<decltype(km * um), meters_squared::value<std::milli, double>>);
+
+    static_assert(std::same_as<decltype(um * um), meters_squared::value<std::pico, double>>);
+
+    static_assert(std::same_as<decltype(m / m), double>);
+    // static_assert(std::same_as<decltype(m / km), double>); // fails compiling on purpose
```

Comment:
Consider describing why, so if we change behaviour or semantics in the future we can figure out if it broke an assumption or made new things possible.